internal/entitlement/entitlement.go

```diff
@@ -84,14 +87,46 @@ type GenericProperties struct {
 	SubjectKey      string          `json:"subjectKey,omitempty"`
 	EntitlementType EntitlementType `json:"type,omitempty"`
 
-	UsagePeriod *UsagePeriod `json:"usagePeriod,omitempty"`
+	UsagePeriod        *UsagePeriod `json:"usagePeriod,omitempty"`
+	CurrentUsagePeriod *api.Period  `json:"currentUsagePeriod,omitempty"`
 }
 
 type UsagePeriod struct {
 	Anchor   time.Time           `json:"anchor"`
 	Interval UsagePeriodInterval `json:"interval"`
 }
 
+func (u UsagePeriod) ToRecurringPeriod() api.RecurringPeriod {
```

Comment:
i don't like api being referred in the core internal package, lets do the mapping in the adapter/driver layer